core/src/scene/styleContext.cpp

```diff
@@ -25,20 +24,8 @@ static const std::string key_zoom("$zoom");
 StyleContext::StyleContext() {
     m_ctx = duk_create_heap_default();
 
-    // add empty feature_object
-    duk_push_object(m_ctx);
-
-    // assign instance to feature_object
-    duk_push_pointer(m_ctx, this);
-    if (!duk_put_prop_string(m_ctx, -2, DATA_ID)) {
-        LOGE("Ctx not assigned");
-    }
-
-    // put object in global scope
-    if (!duk_put_global_string(m_ctx, "feature")) {
-        LOGE("Feature not assigned");
-    }
-
+    //// Create global geometry constants
+    // TODO make immutable
```

Comment:
It seems that these globals are not strictly immutable in tangram-js, so to most nearly reproduce its behavior we should technically allow these to be re-assigned within a function. However tangram-js also confines these globals to a closure for each function evaluation, so re-assignment would not impact their value in other functions. It seems that we would need to assign the global variables before every evaluation in order to mimic this. 

Since this is a relatively unlikely case, I'm fine postponing its resolution until we run out of more important issues to work on, but now at least we know about this potential discrepancy.